src/node_api_embedding.cc

```diff
@@ -0,0 +1,765 @@
+  }
+  result |= node::ProcessInitializationFlags::kNoInitializeV8;
+  result |= node::ProcessInitializationFlags::kNoInitializeNodeV8Platform;
+  if ((flags & node_api_platform_no_init_openssl) != 0) {
+    result |= node::ProcessInitializationFlags::kNoInitOpenSSL;
+  }
+  if ((flags & node_api_platform_no_parse_global_debug_variables) != 0) {
+    result |= node::ProcessInitializationFlags::kNoParseGlobalDebugVariables;
+  }
+  if ((flags & node_api_platform_no_adjust_resource_limits) != 0) {
+    result |= node::ProcessInitializationFlags::kNoAdjustResourceLimits;
+  }
+  if ((flags & node_api_platform_no_use_large_pages) != 0) {
+    result |= node::ProcessInitializationFlags::kNoUseLargePages;
+  }
+  if ((flags & node_api_platform_no_print_help_or_version_output) != 0) {
+    result |= node::ProcessInitializationFlags::kNoPrintHelpOrVersionOutput;
+  }
+  if ((flags & node_api_platform_generate_predictable_snapshot) != 0) {
+    result |= node::ProcessInitializationFlags::kGeneratePredictableSnapshot;
+  }
+  return static_cast<node::ProcessInitializationFlags::Flags>(result);
+}
+
+node::EnvironmentFlags::Flags GetEnvironmentFlags(node_api_env_flags flags) {
+  uint64_t result = node::EnvironmentFlags::kNoFlags;
+  if ((flags & node_api_env_default_flags) != 0) {
+    result |= node::EnvironmentFlags::kDefaultFlags;
+  }
+  if ((flags & node_api_env_owns_process_state) != 0) {
+    result |= node::EnvironmentFlags::kOwnsProcessState;
+  }
+  if ((flags & node_api_env_owns_inspector) != 0) {
+    result |= node::EnvironmentFlags::kOwnsInspector;
+  }
+  if ((flags & node_api_env_no_register_esm_loader) != 0) {
+    result |= node::EnvironmentFlags::kNoRegisterESMLoader;
+  }
+  if ((flags & node_api_env_track_unmanaged_fds) != 0) {
+    result |= node::EnvironmentFlags::kTrackUnmanagedFds;
+  }
+  if ((flags & node_api_env_hide_console_windows) != 0) {
+    result |= node::EnvironmentFlags::kHideConsoleWindows;
+  }
+  if ((flags & node_api_env_no_native_addons) != 0) {
+    result |= node::EnvironmentFlags::kNoNativeAddons;
+  }
+  if ((flags & node_api_env_no_global_search_paths) != 0) {
+    result |= node::EnvironmentFlags::kNoGlobalSearchPaths;
+  }
+  if ((flags & node_api_env_no_browser_globals) != 0) {
+    result |= node::EnvironmentFlags::kNoBrowserGlobals;
+  }
+  if ((flags & node_api_env_no_create_inspector) != 0) {
+    result |= node::EnvironmentFlags::kNoCreateInspector;
+  }
+  if ((flags & node_api_env_no_start_debug_signal_handler) != 0) {
+    result |= node::EnvironmentFlags::kNoStartDebugSignalHandler;
+  }
+  if ((flags & node_api_env_no_wait_for_inspector_frontend) != 0) {
+    result |= node::EnvironmentFlags::kNoWaitForInspectorFrontend;
+  }
+  return static_cast<node::EnvironmentFlags::Flags>(result);
+}
+
+}  // end of anonymous namespace
+}  // end of namespace v8impl
+
+napi_status NAPI_CDECL
+node_api_initialize_platform(int32_t argc,
```

Comment:
should we have this return an opaque handle that is passed into dispose?  Just wondering if we should handle the platform through a reference versus treating it like a global, even if that is the case today?

Reply:
I specifically changed it the other way around. IMHO, having the opaque handle for the platform creates a false impression that we can have multiple platforms. I found it to be confusing. We can always add the multiplatform feature in future if it is ever become available.

Reply:
k, makes sense.

Reply:
In the new API iteration, I had restored the platform opaque object.